src/Orleans/Streams/Internal/StreamConsumerExtension.cs

```diff
@@ -5,13 +5,14 @@
 using System.Threading.Tasks;
 using Orleans.Concurrency;
 using Orleans.Runtime;
+using Orleans.Serialization;
 
 namespace Orleans.Streams
 {
     internal interface IStreamSubscriptionHandle
     {
         Task<StreamHandshakeToken> DeliverItem(object item, StreamSequenceToken currentToken, StreamHandshakeToken handshakeToken);
-        Task<StreamHandshakeToken> DeliverBatch(IBatchContainer item, StreamHandshakeToken handshakeToken);
+        Task<StreamHandshakeToken> DeliverBatch(IBatchContainer item, StreamHandshakeToken handshakeToken, SerializationManager serializationManager);
```

Comment:
hmmm, but that's promising, if there's only a few places where the serialization manager is used, then perhaps we can drop the `new()` constraint on them, and use `ActivatorUtilities` to new them up, and only the ones that actually care about SerializationManager (which we currently have, but expect to get rid at some point) can actually get injected with it in their constructors...
It's probably harder than that in practice, but something to explore...